// homey-api/lib/HomeyAPI/HomeyAPIV2/ManagerInsights.js
'use strict';

const ManagerInsightsV3 = require('../HomeyAPIV3/ManagerInsights');
const Log = require('./ManagerInsights/Log');

class ManagerInsights extends ManagerInsightsV3 {
  static CRUD = {
    ...super.CRUD,
    Log,
  };

  async getLog({ $cache = true, id, ...rest }) {
    if ($cache === true && this.__cache['log'][id]) {
      return this.__cache['log'][id];
    }

    return this.__super__getLog({
      ...rest,
      id: id.split(':').reverse()[0],
      uri: id.split(':', 3).join(':'),
    });
  }

  async getLogEntries({ id, resolution, ...rest }) {
    const result = await this.__super__getLogEntries({
      ...rest,
      id: id.split(':').reverse()[0],
      uri: id.split(':', 3).join(':'),
      resolution,
    });

    return {
      ...result,
      id: id,
    };
  }

  async deleteLogEntries({ id, ...rest }) {
    return this.__super__deleteLogEntries({
      ...rest,
      id: id.split(':').reverse()[0],
      uri: id.split(':', 3).join(':'),
    });
  }

  // deleteLog
  // updateLog
}

module.exports = ManagerInsights;
